libs/wingc/src/type_check.rs

```diff
@@ -2194,11 +2158,17 @@ impl<'a> TypeChecker<'a> {
 				}
 
 				// Add the constructor to the class env
-				let constructor_type = self.resolve_type_annotation(
+				let mut constructor_type = self.resolve_type_annotation(
 					&TypeAnnotation::FunctionSignature(constructor.signature.clone()),
 					env,
 					stmt.idx,
 				);
+				// use the class type as the constructor's "this" type
+				if let Type::Function(ref mut f) = *constructor_type {
+					f.this_type = Some(class_type.clone());
+				} else {
+					panic!("Expected constructor type to be a function");
+				}
```

Comment:
I'd like to treat the constructor is a static with `this` being something special that's available in it's env, but isn't part of the sig. Lets remove this.